Explorer/Assets/DCL/SDKComponents/AudioStream/AudioStreamComponent.cs

```diff
@@ -0,0 +1,84 @@
+﻿using DCL.ECSComponents;
+using DCL.Optimization.Pools;
+using DCL.Utilities.Extensions;
+using RenderHeads.Media.AVProVideo;
+using System;
+using System.Runtime.CompilerServices;
+
+namespace DCL.SDKComponents.AudioStream
+{
+    public struct AudioStreamComponent : IPoolableComponentProvider<MediaPlayer>
+    {
+        private const float DEFAULT_VOLUME = 1f;
+        private static IComponentPool<MediaPlayer> mediaPlayerPool;
+
+        private string url;
+
+        private MediaPlayer mediaPlayer { get; }
+
+        MediaPlayer IPoolableComponentProvider<MediaPlayer>.PoolableComponent => mediaPlayer;
+
+        Type IPoolableComponentProvider<MediaPlayer>.PoolableComponentType => typeof(MediaPlayer);
+
+        public AudioStreamComponent(PBAudioStream sdkComponent, IComponentPoolsRegistry poolsRegistry, bool isCurrentScene)
+        {
+            mediaPlayerPool ??= poolsRegistry.GetReferenceTypePool<MediaPlayer>();
+            mediaPlayer = mediaPlayerPool.Get();
+
+            url = sdkComponent.Url;
+            mediaPlayer = mediaPlayer;
+
+            UpdateVolume(sdkComponent, isCurrentScene);
+
+            if (sdkComponent.Url.IsValidUrl())
+            {
+                mediaPlayer.OpenMedia(MediaPathType.AbsolutePathOrURL, sdkComponent.Url, autoPlay: false);
+
+                if (sdkComponent is { HasPlaying: true, Playing: true })
+                    mediaPlayer.Play();
+            }
+        }
+
+        public void Dispose()
+        {
+            mediaPlayer.CloseCurrentStream();
+            mediaPlayerPool = null;
+        }
+
+        public void UpdateComponentChange(PBAudioStream sdkComponent)
```

Comment:
I have a feeling that these `UpdateXXX` functions belong to the system rather because they manipulate `SDKComponent` as well and it's a responsibility issue for `AudioStreamComponent` to have this knowledge

Reply:
it was mainly to encapsulate `MediaPlayer` (and other fields) inside the `AudioStreamComponent ` and just expose more high-level interface to operate with it. Otherwise System would need to know all low-level details on how to operate with MediaPlayer (AVPRo) fields. Taking it into account, should I still move this logic into the `AudioStreamSystem`?

Reply:
ok, I still need to expose MediaPlayer for proper cleanUp on sdkComponent removal, so I will move logic into the system